test/integration/point_cloud.cc

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+#include <string>
+
+#include <gz/msgs/float_v.pb.h>
+#include <gz/msgs/marker.pb.h>
+#include <gz/msgs/material.pb.h>
+#include <gz/msgs/pointcloud_packed.pb.h>
+#include <gz/msgs/world_stats.pb.h>
+
+#include <gz/common/Console.hh>
+#include <gz/common/Filesystem.hh>
+#include <gz/msgs/PointCloudPackedUtils.hh>
+#include <gz/msgs/Utility.hh>
+#include <gz/rendering/RenderEngine.hh>
+#include <gz/rendering/RenderingIface.hh>
+#include <gz/rendering/Scene.hh>
+#include <gz/transport/Node.hh>
+#include <gz/utils/ExtraTestMacros.hh>
+
+#include "test_config.hh"  // NOLINT(build/include)
+#include "gz/gui/Application.hh"
+#include "gz/gui/GuiEvents.hh"
+#include "gz/gui/MainWindow.hh"
+#include "gz/gui/Plugin.hh"
+
+int g_argc = 1;
+char* g_argv[] =
+{
+  reinterpret_cast<char*>(const_cast<char*>("./PointCloud_TEST")),
+};
+
+using namespace std::chrono_literals;
+
+using namespace gz;
+using namespace gui;
+
+class PointCloudTestFixture : public ::testing::Test
+{
+  public:
+    gz::transport::Node node;
+    rendering::ScenePtr scene;
+    gz::transport::Node::Publisher pointcloudPub;
+    gz::transport::Node::Publisher flatPub;
+    gz::msgs::PointCloudPacked pcMsg;
+    gz::msgs::Float_V flatMsg;
+    std::atomic<bool> receivedMsg;
+
+    PointCloudTestFixture()
+    {
+      // Periodic world statistics
+      this->pointcloudPub = this->node.Advertise<gz::msgs::PointCloudPacked>(
+        "/point_cloud");
+      this->flatPub = this->node.Advertise<gz::msgs::Float_V>("/flat");
+
+      this->InitMockData();
+
+      this->node.Advertise("/marker",
+        &PointCloudTestFixture::OnMarkerMsg, this);
+      this->receivedMsg = false;
+    }
+
+    public: void InitMockData()
+    {
+      gz::msgs::InitPointCloudPacked(pcMsg, "some_frame", true,
```

Comment:
nit: adding `this->` to member variables (below as well)

```suggestion
      gz::msgs::InitPointCloudPacked(this->pcMsg, "some_frame", true,
```